client/my-sites/plugins/access-control.js

```diff
@@ -6,32 +6,23 @@ import i18n from 'i18n-calypso';
 /**
  * Internal Dependencies
  */
-import sitesList from 'lib/sites-list';
 import config from 'config';
 import notices from 'notices';
 
-const sites = sitesList();
-
-const hasErrorCondition = ( site, type ) => {
-	const errorConditions = {
-		notMinimumJetpackVersion: site && ! site.hasMinimumJetpackVersion && site.jetpack,
-		notRightsToManagePlugins: sites.initialized && ! sites.canManageSelectedOrAll()
-	};
-	return errorConditions[ type ];
+const getHasRightsToManagePlutins = ( selectedOrAll ) => {
```

Comment:
As you noted, this would probably work better as two selectors. If we're iterating through all sites, I would recommend memoizing with https://github.com/Automattic/wp-calypso/tree/master/client/lib/create-selector